src/webgpu/api/validation/encoding/cmds/render/indirect_multi_draw.spec.ts

```diff
@@ -0,0 +1,387 @@
+      size: 256,
+      usage: indirectUsage,
+    });
+    const drawCountBuffer = useDrawCountBuffer
+      ? t.createBufferTracked({
+          size: 256,
+          usage: drawCountUsage,
+        })
+      : undefined;
+
+    const { encoder, validateFinish } = t.createEncoder('render pass');
+    encoder.setPipeline(t.createNoOpRenderPipeline());
+    if (indexed) {
+      encoder.setIndexBuffer(t.makeIndexBuffer(), 'uint32');
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndexedIndirect(indirectBuffer, 0, 1, drawCountBuffer);
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndirect(indirectBuffer, 0, 1, drawCountBuffer);
+    }
+    const shouldSucceed =
+      (indirectUsage & GPUBufferUsage.INDIRECT) !== 0 &&
+      (!drawCountBuffer || drawCountUsage & GPUBufferUsage.INDIRECT) !== 0;
+    validateFinish(shouldSucceed);
+  });
+
+g.test('offsets_alignment')
+  .desc(
+    `
+Tests indirect and draw count offsets must be a multiple of 4.
+  `
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase('chromium-experimental-multi-draw-indirect' as GPUFeatureName);
+  })
+  .paramsSubcasesOnly(
+    kIndirectMultiDrawTestParams
+      .combine('indirectOffset', [0, 2, 4] as const)
+      .combine('drawCountOffset', [0, 2, 4] as const)
+  )
+  .fn(t => {
+    const { indexed, indirectOffset, useDrawCountBuffer, drawCountOffset } = t.params;
+
+    const indirectBuffer = t.createBufferTracked({
+      size: 256,
+      usage: GPUBufferUsage.INDIRECT,
+    });
+    const drawCountBuffer = useDrawCountBuffer
+      ? t.createBufferTracked({
+          size: 256,
+          usage: GPUBufferUsage.INDIRECT,
+        })
+      : undefined;
+
+    const { encoder, validateFinish } = t.createEncoder('render pass');
+    encoder.setPipeline(t.createNoOpRenderPipeline());
+    if (indexed) {
+      encoder.setIndexBuffer(t.makeIndexBuffer(), 'uint32');
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndexedIndirect(
+        indirectBuffer,
+        indirectOffset,
+        1,
+        drawCountBuffer,
+        drawCountOffset
+      );
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (encoder as any).multiDrawIndirect(
+        indirectBuffer,
+        indirectOffset,
+        1,
+        drawCountBuffer,
+        drawCountOffset
+      );
+    }
+
+    // We need to figure out if https://github.com/gpuweb/gpuweb/pull/2315/files#r1773031950 applies.
+    validateFinish(indirectOffset % 4 === 0 && (!useDrawCountBuffer || drawCountOffset % 4 === 0));
+  });
+
+interface IndirectOffsetOobCase {
+  indirectOffset: number;
+  bufferSize: number;
+  useDrawCountBuffer?: boolean;
+  drawCountOffset?: number;
+  maxDrawCount?: number;
+  _valid: boolean;
+}
+
+g.test('indirect_offset_oob')
```

Comment:
Test name implies it only tests OOB on the indirectBuffer, but is also doing a bunch of stuff with drawCountBuffer that is hard to separate in my head. Can there be two tests:
- `indirectBuffer_OOB` (just a single boolean param for whether it uses a drawCountBuffer or not)
- `drawCountBuffer_OOB`
?

Reply:
I've renamed this test as the implementation of two tests would be identical.

Reply:
I'd prefer if the two were totally separate, as the validation for OOB on each of the two buffers is totally orthogonal. For offsets_alignment it's fine because the list of cases is small, but this list of cases is large and needs to test all the corner cases of two separate checks. Reviewing the list to make sure it covers all of those cases is difficult.

If I'm misunderstanding and they're not orthogonal then this is fine, I just still will need to review the list of cases more closely.

Reply:
(Also the other tests use kIndirectMultiDrawTestParams so if they were split I think they would have a lot of duplicated control cases; that's not the case here)

Reply:
Done.